bin/yii

```diff
@@ -10,42 +10,29 @@ use Yiisoft\Yii\Console\Application;
 use Yiisoft\Yii\Console\Output\ConsoleBufferedOutput;
 
 (static function () {
-    $cwd = getcwd();
-
-    $possibleAutoloadPaths = [
-        // running from project root
-        $cwd . '/vendor/autoload.php',
-        // running from project bin
-        dirname($cwd) . '/autoload.php',
-        // local dev repository
-        dirname(__DIR__) . '/vendor/autoload.php',
-        // dependency
-        dirname(__DIR__, 4) . '/vendor/autoload.php',
-    ];
-
     $autoloadPath = null;
 
-    foreach ($possibleAutoloadPaths as $possibleAutoloadPath) {
+    for ($x = 5; $x > 0; $x--) {
+        $possibleAutoloadPath = dirname(__FILE__, $x) . '/vendor/autoload.php';
         if (file_exists($possibleAutoloadPath)) {
             $autoloadPath = $possibleAutoloadPath;
             break;
         }
     }
```

Comment:
```suggestion
    $autoloadPath = null;

    if (basename(dirname(__DIR__)) === 'yii-console') {
        // yii-dev special case
        $cwd = getcwd();

        $possibleAutoloadPaths = [
            // running from project root
            $cwd . '/vendor/autoload.php',
            // running from project bin
            dirname($cwd) . '/autoload.php',
        ];

        foreach ($possibleAutoloadPaths as $possibleAutoloadPath) {
            if (file_exists($possibleAutoloadPath)) {
                $autoloadPath = $possibleAutoloadPath;
                break;
            }
        }
    }

    if ($autoloadPath === null) {
        for ($levels = 1; $levels <= 5; $levels++) {
            $possibleAutoloadPath = dirname(__FILE__, $levels) . '/vendor/autoload.php';
            if (file_exists($possibleAutoloadPath)) {
                $autoloadPath = $possibleAutoloadPath;
                break;
            }
        }
    }
```

Reply:
Unfortunately, it is impossible to solve it w/o `getcwd()`. Related to https://github.com/yiisoft/yii-demo/issues/112